src/cls/rbd/cls_rbd.cc

```diff
@@ -1678,7 +1678,9 @@ int snapshot_add(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
 	        (unsigned long long)snap_id.val);
 	return -EIO;
       }
-      if (snap_meta.name == old_meta.name || snap_meta.id == old_meta.id) {
+      if ((snap_meta.name == old_meta.name
+	    && snap_meta.snapshot_namespace == old_meta.snapshot_namespace)
```

Comment:
Minor: the conditionals should go at the end of the line and not the start of a new line:

```
if ((snap_meta.name == old_meta.name &&
     snap_meta.snapshot_namespace == old_meta.snapshot_namespace) ||
    snap_meta.id == old_meta.id) {
```